projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -837,6 +837,40 @@ function wpcom_launchpad_is_site_launched( $task, $is_complete ) {
 	}
 }
 
+/**
+ * Returns true if one of the site's WooCommerce tasks is complete.
+ *
+ * @param Task $task The task object.
+ * @param bool $is_complete The current task status.
+ *
+ * @return boolean
+ */
+function wpcom_launchpad_is_woocommerce_task_completed( $task, $is_complete ) {
+	if ( $is_complete ) {
+		return true;
+	}
+
+	// Array mapping task 'id' from $task to keys used in 'woocommerce_task_list_tracked_completed_tasks' site option
+	$task_map = array(
+		'woo_customize_store'      => 'customize-store',
+		'woo_products'             => 'products',
+		'woo_woocommerce_payments' => 'woocommerce-payments',
+		'woo_tax'                  => 'tax',
+		'woo_marketing'            => 'marketing',
+		'woo_add_domain'           => 'add_domain',
+		'woo_launch_site'          => 'launch_site',
+	);
+
+	$completed_tasks = get_option( 'woocommerce_task_list_tracked_completed_tasks', array() );
+
+	if ( array_key_exists( $task['id'], $task_map ) && in_array( $task_map[ $task['id'] ], $completed_tasks, true ) ) {
+		wpcom_mark_launchpad_task_complete( $task['id'] );
```

Comment:
Not sure if marking it as complete on launchpad end is needed, and if it would cause disparity between woo's task list and launchpad's task list. But for now it is fine.

Reply:
Hmm. Good point. I think a disparity could happen if, for some reason, a task on the Woo side is uncompleted. The reason for this is that we are not handling task un-completing on the Calypso side in any way at the moment.

What you are suggesting is to get the current completion status from each Woo task without completing the task on the Calypso side, right?

I have created a dedicated task for it: https://github.com/Automattic/dotcom-forge/issues/6893